n3fit/src/evolven3fit/__init__.py

```diff
@@ -0,0 +1,239 @@
+"""
+The evolven3fit_API module provides the necessary tools to evolve a fit 
+using eko and to prepare it for postfit.  
+"""
+
+import pathlib
+import logging
+import numpy as np
+import yaml
+import os
+import glob
+
+from ekobox import gen_theory, gen_op, genpdf, gen_info
+from ekomark import apply
+from eko import basis_rotation as br
+from eko import run_dglap
+from eko import output
+
+from . import utils
+from validphys.loader import Loader
+
+
+logger = logging.getLogger(__name__)
+
+
+def evolve_fit(conf_folder, op_card_dict, t_card_dict, eko_path=None, dump_eko=None):
+    """
+    Evolves all the fitted replica in conf_folder/nnfit
+
+    Parameters
+    ----------
+
+        conf_folder: str or pathlib.Path
+            path to the folder containing the fit
+        op_card_dict: dict
+            user settings for the op_card
+        t_card_dict: dict
+            user settings for the t_card
+        eko_path: str or pathlib.Path
+            path where the eko is stored (if None the eko will be
+            recomputed)
+        dump_eko: str or pathlib.Path
+            path where the eko is dumped (if None the eko won't be
+            stored)
+    """
+    log_file = pathlib.Path(conf_folder) / "evolven3fit.log"
+    if log_file.exists():
+        raise SystemError("Log file already exists: evolven3fit has already been run?")
+    log_file = logging.FileHandler(log_file)
+    log_file.setLevel(logging.INFO)
+    log_file.setFormatter(
+        logging.Formatter("%(asctime)s %(name)s/%(levelname)s: %(message)s")
+    )
+    for logger_ in (logger, *[logging.getLogger("eko")]):
+        logger_.handlers = []
+        logger_.setLevel(logging.INFO)
+        logger_.addHandler(log_file)
+    usr_path = pathlib.Path(conf_folder)
+    initial_PDFs_dict = load_fit(usr_path)
+    theory, op = construct_eko_cards(usr_path, op_card_dict, t_card_dict)
+    if eko_path is not None:
+        eko_path = pathlib.Path(eko_path)
+        logger.info(f"Loading eko from : {eko_path}")
+        eko_op = output.Output.load_tar(eko_path)
+    else:
+        eko_op = construct_eko_for_fit(theory, op, dump_eko)
+    info = gen_info.create_info_file(theory, op, 1, info_update={})
+    info["NumMembers"] = "REPLACE_NREP"
+    info["ErrorType"] = "replicas"
+    dump_info_file(usr_path, info)
+    utils.fix_info_path(usr_path)
+    for replica in initial_PDFs_dict.keys():
+        evolved_block = evolve_exportgrid(
+            initial_PDFs_dict[replica], eko_op, theory, op
+        )
+        dump_evolved_replica(
+            evolved_block, usr_path, int(replica.removeprefix("replica_"))
+        )
+        # fixing_replica_path
+        utils.fix_replica_path(usr_path, int(replica.removeprefix("replica_")))
+    # remove folder
+    (usr_path / "nnfit" / usr_path.stem).rmdir()
```

Comment:
I am not too sure what is being removed here and why are we doing that. Could you elaborate, including in the comment. My sense is that this script should not be in the business of deleting (or moving around) any existing inputs.

Reply:
But the function coming from banana is used within the previous call `dump_evolved_replica`.
Since `dump_evolved_replica` already takes as input all the information `fix_replica_path` needs, the `fix_replica_path` functionality should be part of `dump_evolved_replica` which in practice means that `dump_evolved_replica` dumps the information in the right place (even if it silently creates temporary files).

(even better if those temporary files were created in `/tmp/` but if not possible at this point that's ok)

Reply:
Ah ok then yes, I already have included the fix funcs in the dump funcs for both the info file and replica exportgrids :)